### PR TITLE
additional rollups

### PR DESCRIPTION
**Pull request by z3dev**
These changes add rollup configurations to packages, allowing both UMD and ES ready packages to be built. The focus is on packages that are competing all tests and stable.

For the IO packages, only the modeling library is expected to be 'external'. The other dependencies are rolled up into the IO package. For example, the svg-deserializer library includes array-utils and saxes packages.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

### CODE REVIEW

**Comment by z3dev** [comment]
@platypii @Loosetooth can you take a look at these initial rollup configurations. it's all new ground for me so some feedback would be good BEFORE making additional changes.

**Comment by z3dev** [comment]
@hrgdavor @platypii can you review these changes quickly? this is the next step towards publishing V3 libraries.

let me know if any other libraries need packaging. i'm actually starting to understand rollup now. ;)

**Comment by hrgdavor** [comment]
@z3dev what would be a good way to test ? try to build ?

As always, I also have something unrelated too. 

One thing I discovered while trying to make vscode to work nicely with my libraries is to include source in npm publishing, and not just bundled cjs and es.
https://docs.npmjs.com/cli/v9/configuring-npm/package-json#files

this way I do not need to produce d.ts files to have nice code hints in vscode, and I can navigate library code from within a project that uses the library. That is very much in-line with how I navigate source code of libraries when coding in Java. Many times I can get more useful info from source than from the documentation.

**Comment by platypii** [comment]
I also have questions of how to run / test this. I was able to build bundles by doing:

```
pnpm i
cd packages/io/x3d-deserializer
npm run build
```

I don't know if this is how @z3dev expected it to be run though, it seems like there should be a better way.

File: packages/io/3mf-serializer/package.json [hunk 1]
```diff
@@ -34,8 +37,12 @@
     "fflate": "0.7.3"
   },
   "devDependencies": {
+    "@rollup/plugin-commonjs": "^23.0.4",
+    "@rollup/plugin-node-resolve": "^15.0.1",
```
Comment (platypii) [inline packages/io/3mf-serializer/package.json:41]: Why does the 3mf-serializer depend on these two plugins, but most other packages don't?

File: packages/io/x3d-deserializer/rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      file: 'dist/jscad-x3d-deserializer.es.js',
+      format: 'es',
+    }
+  ],
+
+  plugins: [
+    commonjs(),
+    banner('<%= pkg.description %>\n<%= pkg.name %>\nVersion <%= pkg.version %>\n<%= pkg.license %> License'),
+    versionInjector({injectInComments:{fileRegexp: /\.(html)$/},logLevel: 'warn'}),
+    nodeResolve()
```
Comment (platypii) [inline packages/io/x3d-deserializer/rollup.config.js:30]: Why does `x3d-deserializer` depend on these 4 plugins, but for example, `dxf-deserializer` only depends on 2 plugins?

File: packages/io/json-deserializer/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+export default {
+  input: 'src/index.js',
+
+  output: [
+    {
+      file: 'dist/jscad-json-deserializer.min.js',
+      format: 'umd',
+      name: 'jscadJsonDeserializer',
+    },
```
Comment (platypii) [inline packages/io/json-deserializer/rollup.config.js:13]: Most other io packages have a global `jscadModeling` export. Why the inconsistency? I would prefer to not use globals if not necessary.

File: packages/io/3mf-serializer/package.json (hunk 1 above)
Comment (z3dev) [reply packages/io/3mf-serializer/package.json:41]: So, the 3mf serializer depends on a library called "", which is CommonJS (require based imports). Node of course understands both CommonJS and ES6 imports. But Rollup needs to find modules using plugin-node-resolve, and translate the CommonJS code to ES6 using plugin-commonjs.

Most of the packages depend on modeling and/or array-utils. These are both ES6 modules now. :)

File: packages/io/x3d-deserializer/rollup.config.js (hunk 2 above)
Comment (z3dev) [reply packages/io/x3d-deserializer/rollup.config.js:30]: It has a dependency on 'saxes' with is CommonJS.

File: packages/io/json-deserializer/rollup.config.js (hunk 3 above)
Comment (z3dev) [reply packages/io/json-deserializer/rollup.config.js:13]: Hmm... only depends on array-utils.

**Comment by z3dev** [comment]
> I also have questions of how to run / test this. I was able to build bundles by doing:
> 
> ```
> pnpm i
> cd packages/io/x3d-deserializer
> npm run build
> ```
> 
> I don't know if this is how @z3dev expected it to be run though, it seems like there should be a better way.

Sorry. I haven't plugged the 'build' into lerna yet. There are still a few more things to do before publishing starts.

**Review by platypii — APPROVED** [review]
Seems like reasonable configs. I'm not a rollup expert, but it seems like a good starting point. I'm in favor of merging into V3 branch so we can continue making progress.

**Review by hrgdavor — APPROVED** [review]
looks good :)

**Comment by z3dev** [comment]
@hrgdavor @platypii thanks for the reviews. Once all tests start passing then the publishing of V3 libraries can start. For V2, publishing started as 'alpha' versions. I hope to follow the same with V3.

**Pull request merged**